tsdb/shard.go

```diff
@@ -255,9 +266,7 @@ func (s *Shard) WritePoints(points []models.Point) error {
 			}
 
 			// This was populated earlier, don't need to validate that it's there.
-			s.mu.RLock()
```

Comment:
Can you keep the locks here and explicitly release them for the `ErrEngineClose` check above?  These locks were intentionally placed here to reduce contention as seen under load.